src/main/java/queue/impl/ArrayQueueImpl.java

```diff
@@ -0,0 +1,108 @@
+package queue.impl;
+
+import java.util.*;
+
+public class ArrayQueueImpl<T> extends AbstractQueue<T> {
+
+    private static final int DEFAULT_CAPACITY = 10_001; // Picked by a dice roll =)
+    private static final int CAPACITY_CHANGE_RATE = 2;
+    private int capacity;
+    private final boolean blocking;
+    private Object[] items;
+    private int head;
+    private int tail;
+
+    public ArrayQueueImpl() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public ArrayQueueImpl(int capacity) {
+        this(capacity, true);
+    }
+
+    public ArrayQueueImpl(int capacity, boolean blocking) {
+        this.capacity = capacity;
+        this.items = new Object[capacity];
+        this.blocking = blocking;
+    }
+
+    private void extendCapacity() throws ArithmeticException { // Explicitly notify that we can throw here
+        capacity = Math.multiplyExact(capacity, CAPACITY_CHANGE_RATE);
```

Comment:
You can just multiply here, because I'm sure we don't want to try to create so huge queue )

Reply:
The thing is that we might eventually try to get above `Integer.MAX` and our head/tail pointers would break since those are `int`. By using `Math.multplyExact` we sort of check this case and don't go above `int.max`

Reply:
okay